File: 1-two-sum/twoSum.php
<?php
function twoSum(array $nums, int $target) {
    $numsMap = [];
    foreach ($nums as $idx => $num) {
        $numsMap[$num] = $idx;
        $diffrence = $target - $num;
        if (isset($numsMap[$diffrence])) {
            return [$numsMap[$diffrence], $idx];
        }
    }
    return [];
}

$nums = [11, 2, 6, 7, 9];
$target = 9;
var_dump(twoSum($nums, $target));

File: 1-two-sum/twoSum.go
package main

import "fmt"

func main() {
	nums := []int{11, 7, 2, 5}
	result := towSum(nums, 9)
	fmt.Println(result)
}

func towSum(nums []int, target int) []int {
	var diffrence int
	numsMap := make(map[int]int)
	for i, n := range nums {
		numsMap[n] = i
		diffrence = target - n
		if idx, ok := numsMap[diffrence]; ok {
			return []int{idx, i}
		}
	}
	return []int{}
}
